packages/components/src/internal/components/search/FilterFacetedSelector.tsx

```diff
@@ -1,64 +1,162 @@
+
+        return fieldDistinctValues?.filter(val => {
+            let valuesToKeep = [];
+            if (checkedValues?.indexOf(ALL_VALUE_DISPLAY) === -1) valuesToKeep = checkedValues;
+            return valuesToKeep?.indexOf(val) > -1 || val?.toLowerCase().indexOf(searchStr.toLowerCase()) > -1;
         });
-    }, [activeFieldDistinctValues, selectDistinctOptions]);
+    }, [fieldDistinctValues, searchStr]);
 
     return (
         <>
             {error && <Alert>{error}</Alert>}
-            {!activeFieldDistinctValues && <LoadingSpinner />}
-            <div className="list-group search-parent-entity-col">
-                <ul className="nav nav-stacked labkey-wizard-pills">
-                    {activeFieldDistinctValues?.map((value, index) => {
-                        let displayValue = value;
-                        if (value === null || value === undefined) displayValue = '[blank]';
-                        if (value === true) displayValue = 'TRUE';
-                        if (value === false) displayValue = 'FALSE';
-                        return (
-                            <li key={index}>
-                                <div className="form-check">
-                                    <input
-                                        className="form-check-input"
-                                        type="checkbox"
-                                        name={'field-value-' + index}
-                                        disabled={true}
-                                        checked={false}
-                                    />
-                                    <span style={{ marginLeft: 5 }}>{displayValue}</span>
-                                </div>
-                            </li>
-                        );
-                    })}
-                </ul>
+            {!fieldDistinctValues && <LoadingSpinner />}
+            <div className="search-filter-values__panel">
+                {fieldDistinctValues?.length > showSearchLength && (
+                    <div>
+                        <input
+                            id="find-filter-typeahead-input"
+                            className="form-control find-filter-typeahead-input"
+                            value={searchStr ?? ''}
+                            onChange={onSearchStrChange}
+                            type="text"
+                        />
+                    </div>
+                )}
+                <Row>
+                    <Col xs={6}>
+                        <ul className="nav nav-stacked labkey-wizard-pills">
+                            {filteredFieldDistinctValues?.map((value, index) => {
+                                let displayValue = value;
+                                if (value === null || value === undefined) displayValue = '[blank]';
+
+                                return (
+                                    <li
+                                        key={index}
```

Comment:
are there any extra rerenders happening here because of using `index` for the key? could the `value` be used instead since those should be unique?